src/main/java/org/apache/sling/models/jacksonexporter/impl/JacksonExporter.java

```diff
@@ -58,9 +56,7 @@
 
     private static final int MAPPER_FEATURE_PREFIX_LENGTH = MAPPER_FEATURE_PREFIX.length();
 
-    @Reference(name = "moduleProvider", referenceInterface = ModuleProvider.class,
-            cardinality = ReferenceCardinality.OPTIONAL_MULTIPLE, policy = ReferencePolicy.DYNAMIC)
-    private final RankedServices<ModuleProvider> moduleProviders = new RankedServices<ModuleProvider>(Order.ASCENDING);
+    private final RankedServices<ModuleProvider> moduleProviders = new RankedServices<>(Order.ASCENDING);
```

Comment:
RankedServices are no longer necessary. Field injections are ordered by DS automatically: http://docs.osgi.org/specification/osgi.cmpn/7.0.0/service.component.html#d0e37828

> SCR must set the field value with a new mutable collection that must contain the initial set of bound services sorted using the same ordering as ServiceReference.compareTo based upon service ranking and service id.

Reply:
the correct ordering of services injected here is not validated by any test case.

i've created a separate ticket for this, let us not mix this in this PR: https://issues.apache.org/jira/browse/SLING-10964